azure-kusto-data/source/connectionBuilder.js

```diff
@@ -127,13 +117,11 @@ module.exports = class KustoConnectionStringBuilder {
         return kcsb;
     }
 
-    static withAadManagedIdentities(connectionString, msiEndpoint, msiSecret) {
-        if (!msiEndpoint || msiEndpoint.trim().length == 0) throw new Error("Invalid endpoint");
-        if (!msiSecret || msiSecret.trim().length == 0) throw new Error("Invalid secret");
-
+    static withAadManagedIdentities(connectionString, msiEndpoint = undefined, clientId = undefined) {
         const kcsb = new KustoConnectionStringBuilder(connectionString);
-        kcsb[KeywordMapping.msiEndpoint.propName] = msiEndpoint;
-        kcsb[KeywordMapping.msiSecret.propName] = msiSecret;
+        kcsb.msiEndpoint = msiEndpoint || "http://169.254.169.254/metadata/identity/oauth2/token";
```

Comment:
I don't know if this default is true for all different flavors. Azure function as an endpoing like 'http://127.0.0.1:41092/MSI/token/' when used the environment variable. Also is missing the `secret`.

Reply:
This is strange because they specifically wrote that the localhost endpoint is going away,,I guess it makes sense that for functions you would still need a localhost endpoint.

Anyway, it may not be the default for all services, but it does seem to be the default that is most documented.

You can still provide the endpoint and will still work, or, like I mentions before, I might as well try and get it from env variables. does that make sense?

Reply:
Works for me as long as we're able to provide the secret or as you mentioned before get it from the env variables.